src/main/java/httpserver/HttpServer.java

```diff
@@ -0,0 +1,210 @@
+package httpserver;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+
+/**
+ * HttpServer is a relatively simple class with one job, and one job only:
+ * wait for incoming connections, and send the connections over to an
+ * HttpRequest and an HttpResponse.
+ *
+ * An HttpServer is not required to use the rest of the httpserver classes,
+ * and might not be the best base server for one to use. It exists solely to
+ * provide an existing mechanism for using the rest of the httpserver package.
+ */
+public class HttpServer extends HttpHandler {
```

Comment:
This should probably become a layer on top of something else, like Jetty or Tomcat.

For reference, Spark uses Jetty as its backend.

What I'm thinking specifically is that the current API should remain intact, but the internals should use something more robust and less written by us.